lib/ohai/plugins/linux/network.rb

```diff
@@ -134,6 +138,25 @@ def parse_routes(family, iface)
     end.compact.flatten
   end
 
+  def ethernet_layer_one(iface)
+    return iface unless ethtool_binary = find_ethtool_binary
+    keys = %w[ Speed Duplex Port Transceiver Auto-negotiation MDI-X ]
+    iface.each_key do |tmp_int|
+      next unless iface[tmp_int][:encapsulation] == 'Ethernet'
+      so = shell_out("#{ethtool_binary} #{tmp_int}")
+      so.stdout.lines do |line|
+        line.chomp!
+        Ohai::Log.debug("Parsing ethtool output: #{line}")
+        line.lstrip!
+        k, v = line.split(': ')
+        next unless keys.include? k
+        k.downcase!
+        v.gsub!(/Mb\/s/, '')
```

Comment:
Also speed should probably get turned into a int so that people can do quick comparisons on it.